mapping/PANCAN-mutation/scripts/map-mutations.py

```diff
@@ -0,0 +1,125 @@
+
+# coding: utf-8
+
+# ### Convert mutation gene symbol labels to Entrez IDs  
+# Goal: Relabel the mutation data frame with Entrez IDs instead of gene names, by mapping a combination of chromosome and gene symbol to Entrez ID. The NCBI file downloaded and read in the next cell contains the Entrez ID - gene symbol pairs we will use to do so.
+
+# In[1]:
+
+import os
+import pandas
+from urllib.request import urlretrieve
+import numpy as np
+
+
+# ### Creating Mappings
+
+# In[2]:
+
+base_url = 'ftp://ftp.ncbi.nih.gov/gene/DATA/GENE_INFO/Mammalia/'
+map_name = 'Homo_sapiens.gene_info.gz'
+map_url = base_url + map_name
+path = os.path.join('download', map_name)
+urlretrieve(map_url, path)
+
+path = os.path.join('download', map_name)
+
+map_df = (
+    pandas.read_table(path)
+    .rename(columns={'#tax_id' : 'tax_id', 'GeneID': 'entrez_id', 'chromosome':'chr', 'Symbol':'symbol'})
+)
+
+#filter to include only tax_id == 9606 (human) and columns of interest
+map_df = map_df.ix[map_df['tax_id'] == 9606, ['entrez_id','chr', 'symbol', 'Synonyms']]
+
+map_df.head(2)
+
+
+# In[3]:
+
+map_primary = map_df[['entrez_id','chr', 'symbol']]
+map_primary.head(2)
+
+
+# In[4]:
+
+alternates = map_df['Synonyms'].str.split('|').apply(pandas.Series, 1).stack()
+alternates.name = 'alt_symbol' # needs a name to join
+alternates.index = alternates.index.droplevel(-1)
+map_alternates = map_df.join(alternates)
+
+
+# In[5]:
+
+#format as entrez_id, chr, alternate symbol
+map_alternates = map_alternates.drop_duplicates(subset=['chr','alt_symbol'], keep=False)[
+    ['entrez_id', 'chr', 'alt_symbol']].rename(columns={'alt_symbol': 'symbol'})
+
+
+# In[6]:
+
+#use keep = first to give primacy to the primary symbol convention in the case of ambiguous mappings
+maps_combined = map_primary.append(map_alternates).drop_duplicates(subset=['chr','symbol'], keep='first')
+
+#add chr string to facilitate integration with mutation dataset
+maps_combined['chr'] = 'chr' + maps_combined['chr'].astype(str)
+maps_combined.head(2)
+
+
+# In[7]:
+
+# Check that chr/symbol are all unique 
+assert not (maps_combined.chr + '|' + maps_combined.symbol).duplicated().any()
+
+
+# ### Check and see which mutations in the dataset fail to map
+
+# In[8]:
+
+path = "../../download/PANCAN_mutation.tsv.bz2"
+mutation_df = pandas.read_table(path, index_col=0)
+
+mutation_df.head(2)
+
+
+# In[9]:
+
+failed_mappings = (set(mutation_df.chr + '|' + mutation_df.gene) - 
```

Comment:
Can avoid the string concatenation using tuples.

`set(zip(mutation_df.chr, mutation_df.gene))`